compose/src/main/java/com/bottlerocketstudios/compose/map/GoogleMapsView.kt

```diff
@@ -49,101 +49,141 @@ fun GoogleMapsView(googleMapScreenState: GoogleMapScreenState, toolbarEnabled: B
             MapProperties(maxZoomPreference = MAX_ZOOM_LEVEL, minZoomPreference = MIN_ZOOM_LEVEL)
         )
     }
-    var mapUiSettings by remember {
-        mutableStateOf(MapUiSettings(mapToolbarEnabled = toolbarEnabled))
-    }
+
     // Observing and controlling the camera's state can be done with a CameraPositionState
     val googleCameraPositionState: CameraPositionState = rememberCameraPositionState {
         position = CameraPosition.fromLatLngZoom(googleMapScreenState.dallasLatLng, CITY_ZOOM_LEVEL)
     }
-
     val dialogVisibility = remember {
         mutableStateOf(value = false)
     }
 
+    val fullScreenMaps = remember(googleMapScreenState.businessList) {
+        derivedStateOf { googleMapScreenState.businessList.value.isEmpty() }
+    }
+
+    val showRetryButton = remember {
+        mutableStateOf(value = false)
+    }
+
     dialogVisibility.value = googleMapScreenState.yelpError.value != UserFacingError.NoError
 
     Column(
         modifier = Modifier
             .fillMaxSize()
             .background(Color.DarkGray)
     ) {
+        if (showRetryButton.value) {
+            RetryButton(retry = {
+                showRetryButton.value = false
+                googleMapScreenState.retrySearch()
+            })
+        }
         GoogleMap(
             properties = mapProperties,
-            uiSettings = mapUiSettings,
-            modifier = if (dialogVisibility.value) Modifier.fillMaxSize() else Modifier
+            modifier = if (fullScreenMaps.value) Modifier.fillMaxSize() else Modifier
                 .height(400.dp)
                 .fillMaxWidth(),
             cameraPositionState = googleCameraPositionState
-        )
+        ) {
+            if (googleMapScreenState.googleMarkers.value.isNotEmpty()) {
+                addMarkers(
+                    yelpMarkers = googleMapScreenState.googleMarkers.value,
+                    onclick = { marker ->
+                        Timber.d("${marker.id} ${marker.title}")
+                        googleMapScreenState.setSelectedMarker(marker.tag as YelpMarker)
```

Comment:
In another Story, we'll float the camera control up to view model, this will let us do all sort of cool stuff.